app/src/org/commcare/activities/UpdateActivity.java

```diff
@@ -177,17 +177,22 @@ private void connectToUpdateWorker() {
                                 data.getInt(UpdateWorker.Progress_Total, -1));
                     } else {
                         // Worker getting fired when not running imply completion of worker
-                        ResultAndError<AppInstallStatus> lastUpdateResult = getlastStageUpdateResult();
-                        if (lastUpdateResult != null) {
-                            handleTaskCompletion(getlastStageUpdateResult());
-                        }
+                        handleTaskCompletion(getlastStageUpdateResult());
                     }
                 });
     }
 
     private ResultAndError<AppInstallStatus> getlastStageUpdateResult() {
         UpdateStats updateStats = UpdateStats.loadUpdateStats(CommCareApplication.instance().getCurrentApp());
-        return updateStats.getLastStageUpdateResult();
+        ResultAndError<AppInstallStatus> lastUpdateResult = updateStats.getLastStageUpdateResult();
```

Comment:
Why are we using this one if `ResourceInstallUtils.isUpdateReadyToInstall()` gives the correct result?

Reply:
`ResourceInstallUtils.isUpdateReadyToInstall()` is just an indication of the update's success or failure while in case of failures, the exact reason of the failure is contained in `lastUpdateResult`

Reply:
got it. Thanks for explaining that!